rascal2/dialogs/project_dialog.py

```diff
@@ -0,0 +1,209 @@
+
+    def __init__(self, parent):
+        """
+        Initialize dialog.
+
+        Parameters
+        ----------
+        parent: MainWindowView
+                An instance of the MainWindowView
+        """
+        super().__init__(parent)
+
+        self.setWindowFlags(QtCore.Qt.WindowType.FramelessWindowHint | QtCore.Qt.WindowType.Dialog)
+        self.setModal(True)
+        self.setMinimumWidth(700)
+
+        self.name_error = False
+        self.folder_error = False
+
+        self.create_buttons()
+        self.create_labels()
+        self.add_widgets_to_layout()
+
+    def add_widgets_to_layout(self) -> None:
+        """
+        Add widgets to the layout.
+        """
+        self.main_layout = QtWidgets.QVBoxLayout()
+
+        self.main_layout.setSpacing(20)
+
+        # Add project name widgets
+        layout = QtWidgets.QGridLayout()
+        layout.setVerticalSpacing(2)
+        layout.addWidget(self.project_name_label, 0, 0, 1, 1)
+        layout.addWidget(self.project_name, 0, 1, 1, 5)
+        layout.addWidget(self.project_name_error, 1, 1, 1, 5)
+        self.main_layout.addLayout(layout)
+
+        # Add project folder widgets
+        layout = QtWidgets.QGridLayout()
+        layout.setVerticalSpacing(2)
+        layout.addWidget(self.project_folder_label, 0, 0, 1, 1)
+        layout.addWidget(self.project_folder, 0, 1, 1, 4)
+        layout.addWidget(self.browse_button, 0, 5, 1, 1)
+        layout.addWidget(self.project_folder_error, 1, 1, 1, 4)
+        self.main_layout.addLayout(layout)
+
+        # Add the create and cancel buttons
+        layout = QtWidgets.QHBoxLayout()
+        layout.setAlignment(QtCore.Qt.AlignmentFlag.AlignRight)
+        layout.addWidget(self.create_button)
+        layout.addWidget(self.cancel_button)
+        self.main_layout.addLayout(layout)
+
+        self.setLayout(self.main_layout)
+
+    def create_labels(self) -> None:
+        """
+        Create labels.
+        """
+        # Project name labels
+        self.project_name_label = QtWidgets.QLabel("Project Name:", self)
+        self.project_name_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignLeft)
+        self.project_name_label.setStyleSheet(self._label_style)
+
+        self.project_name = QtWidgets.QLineEdit(self)
+        self.project_name.setPlaceholderText("Enter project name")
+
+        self.project_name_error = QtWidgets.QLabel("Project name needs to be specified.", self)
+        self.project_name_error.setStyleSheet(self._error_style)
+        self.project_name_error.hide()
+
+        # Project folder labels
+        self.project_folder_label = QtWidgets.QLabel("Project Folder:", self)
+        self.project_folder_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignLeft)
+        self.project_folder_label.setStyleSheet(self._label_style)
+
+        self.project_folder = QtWidgets.QLineEdit(self)
+        self.project_folder.setReadOnly(True)
+        self.project_folder.setPlaceholderText("Select project folder")
```

Comment:
it's not necessarily clear from the start that an empty project folder needs to be selected, i can see users selecting a folder then bumping up against the error before they learn that which isn't a super smooth experience. maybe change this to "select empty project folder"?

Reply:
I intentionally left it as three words. I think having both the place holders as the same length looks better. And down the line a non-empty folder can be selected. It just can't have a RasCAL project already saved in it from my understanding. It is also makes the error msgs redundant.

Reply:
fair enough: created issue #22 to track that that's what we want to do down the line